adapter/test/vfd/hermes_vfd_basic_test.cc

```diff
@@ -0,0 +1,718 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Distributed under BSD 3-Clause license.                                   *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Illinois Institute of Technology.                        *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of Hermes. The full Hermes copyright notice, including  *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the top directory. If you do not  *
+ * have access to the file, you may request a copy from help@hdfgroup.org.   *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+using hermes::adapter::vfd::test::MuteHdf5Errors;
+
+/** Returns a number in the range [1, upper_bound] */
+static inline size_t Random1ToUpperBound(size_t upper_bound) {
+  size_t result = ((size_t)GenNextRandom() % upper_bound) + 1;
+
+  return result;
+}
+
+/** Returns a string in the range ["0", "upper_bound") */
```

Comment:
Isn't it `"upper_bound-1"`?

Reply:
I'm using interval notation. The `[` includes 0 on the left, but the `)` on the right means up to but not including `upper_bound`.